File: src/Algorithm/Ⅱ Reduce-and-Conquer/1 insertionSort/insertionSort.cpp
#include <iostream>

using namespace std;

void insertionSort(int a[], int len)
{
    for (int i = 1; i < len; i++)
    {
        int v = a[i];
        int j = i - 1;
        while (j >= 0 && a[j] > v)
        {
            a[j + 1] = a[j];
            j--;
        }
        a[j + 1] = v;
    }
}

int main()
{
    int a[5] = {2, 3, 1, 5, 0};
    insertionSort(a, 5);
    for (int i = 0; i < 5; i++)
    {
        cout << a[i] << endl;
    }
    return 0;
}

File: src/Algorithm/Ⅱ Reduce-and-Conquer/1 insertionSort/insertionSort.js

/**
 * 
 * @param {Array<Number>} a 
 * @returns {Array<Number>} sorted array
 */
function insertionSort(a) {
    
    for (let i = 1; i < a.length; i++){
        let v = a[i];
        let j = i - 1;
        while (j >= 0 && a[j] > v) {
            a[j + 1] = a[j];
            j--;
        }
        a[j + 1] = v;
    }

    return a;
}

var arr = [3,5,1,2,7,8,9,0]

console.log(insertionSort(arr))